modules/iam/main.tf

```diff
@@ -182,84 +211,145 @@ EOF
-                "application-autoscaling:DeregisterScalableTarget",
-                "application-autoscaling:PutScalingPolicy",
-                "application-autoscaling:DeleteScalingPolicy",
-                "application-autoscaling:Describe*"
-            ]
-    }]
+data "aws_iam_policy_document" "segment_emr_service_policy_document" {
+  statement {
+    sid       = "AllowSegmentEMRServicePolicy${var.suffix}"
+    effect    = "Allow"
+    actions   = [
+      "ec2:AuthorizeSecurityGroupEgress",
+      "ec2:AuthorizeSecurityGroupIngress",
+      "ec2:CancelSpotInstanceRequests",
+      "ec2:CreateNetworkInterface",
+      "ec2:CreateSecurityGroup",
+      "ec2:CreateTags",
+      "ec2:DeleteNetworkInterface",
+      "ec2:DeleteSecurityGroup",
+      "ec2:DeleteTags",
+      "ec2:DescribeAvailabilityZones",
+      "ec2:DescribeAccountAttributes",
+      "ec2:DescribeDhcpOptions",
+      "ec2:DescribeImages",
+      "ec2:DescribeInstanceStatus",
+      "ec2:DescribeInstances",
+      "ec2:DescribeKeyPairs",
+      "ec2:DescribeNetworkAcls",
+      "ec2:DescribeNetworkInterfaces",
+      "ec2:DescribePrefixLists",
+      "ec2:DescribeRouteTables",
+      "ec2:DescribeSecurityGroups",
+      "ec2:DescribeSpotInstanceRequests",
+      "ec2:DescribeSpotPriceHistory",
+      "ec2:DescribeSubnets",
+      "ec2:DescribeTags",
+      "ec2:DescribeVpcAttribute",
+      "ec2:DescribeVpcEndpoints",
+      "ec2:DescribeVpcEndpointServices",
+      "ec2:DescribeVpcs",
+      "ec2:DetachNetworkInterface",
+      "ec2:ModifyImageAttribute",
+      "ec2:ModifyInstanceAttribute",
+      "ec2:RequestSpotInstances",
+      "ec2:RevokeSecurityGroupEgress",
+      "ec2:RunInstances",
+      //      "ec2:TerminateInstances",
+      //      "ec2:DeleteVolume",
+      "ec2:DescribeVolumeStatus",
+      "ec2:DescribeVolumes",
+      //      "ec2:DetachVolume",
+      "iam:GetRole",
+      "iam:GetRolePolicy",
+      "iam:ListInstanceProfiles",
+      "iam:ListRolePolicies",
+      "iam:PassRole",
+      "s3:CreateBucket",
+      //      "s3:Get*",
+      //      "s3:List*",
+      "sdb:BatchPutAttributes",
+      "sdb:Select",
+      //      "sqs:CreateQueue",
+      //      "sqs:Delete*",
+      //      "sqs:GetQueue*",
+      //      "sqs:PurgeQueue",
+      //      "sqs:ReceiveMessage",
+      "cloudwatch:PutMetricAlarm",
+      "cloudwatch:DescribeAlarms",
+      "cloudwatch:DeleteAlarms",
+      "application-autoscaling:RegisterScalableTarget",
+      "application-autoscaling:DeregisterScalableTarget",
+      "application-autoscaling:PutScalingPolicy",
+      "application-autoscaling:DeleteScalingPolicy",
+      "application-autoscaling:Describe*"
+    ]
+    resources = [
+      "*",
+    ]
+  }
+
+  statement {
+    actions   = [
+      "ec2:DeleteVolume",
+      "ec2:TerminateInstances",
+      "ec2:DetachVolume",
+    ]
+    resources = [
+      "*"
+    ]
+    # Check that were created by Segment
```

Comment:
Here is where I need your collaboration. Can we filter these actions to specific tagged resources?

Reply:
So the role to which this policy is attached is the EMR service-linked role that is only assumed by the EMR service to perform administrative tasks. You can see that in the[ assume-role section ](https://github.com/segmentio/terraform-aws-data-lake/blob/a298c01cb16c3d4c2c8a6df8c64c5f8ce53aed9c/modules/iam/main.tf#L195-L209)of the `segment_emr_service_role` role definition. Data Lakes directly nor the jobs running on the EMR cluster use this role. 

We have used the default policy for this role given that this is only used by the EMR service and it's not clear what the effect of limiting the access here would be. You can read more about this role [here](https://docs.aws.amazon.com/emr/latest/ManagementGuide/using-service-linked-roles.html).

These are the  roles used by Data lakes to access your resources and we can tighten the permissions for this if required. We have already limited access to an extent but adding some explicit `Deny` rules makes sense.
```
segment_data_lake_iam_role
segment_emr_instance_profile_role
```